packages/http-client-csharp/generator/Microsoft.Generator.CSharp.ClientModel/src/Providers/ScmMethodProviderCollection.cs

```diff
@@ -133,6 +165,38 @@ private IReadOnlyList<ValueExpression> GetParamConversions(IReadOnlyList<Paramet
             return conversions;
         }
 
+        private IReadOnlyList<ValueExpression> GetSpreadParamModelCtorArgs(ConstructorSignature modelProviderCtorSignature)
+        {
+            var convenienceMethodParams = ConvenienceMethodParameters.ToDictionary(p => p.Name);
```

Comment:
This looks like we are adding a new set of methods that don't fit into the old paradigm.

Before we delegated param creation to RestClientProvider and got conv vs protocol via a flag to that method.
We had another method which created a conversion expression to be able to pass the convenience params into the protocol method

Can we try to fit the functionality for these new methods into that paradigm instead?

Reply:
I tried to do some consolidating here but things got a bit messy. This method's intention is to build the arguments to the spread model's ctor so we can declare an instance of that model without assuming the order of the constructor's parameters. This should all still happen when we build the stack vars for the protocol method param conversion.